### PR TITLE
🌱 Use shellcheck binary instead of self-built docker image

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

This PR:

* replaces the current shellcheck implementation (based on a self-built container image used via `docker run`) by the solution which is used at https://github.com/kubernetes-sigs/cluster-api.
* enables to run `make lint-shell` or `make verify-shellcheck` on arm based macs.
* this also adds the `verify-shellcheck` target for compatibility with CAPI's Makefile.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2125

/area ci

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -326,8 +328,7 @@ lint-markdown: ## Lint the project's markdown
 	docker run --rm -v "$$(pwd)":/build$(DOCKER_VOL_OPTS) gcr.io/cluster-api-provider-vsphere/extra/mdlint:0.17.0 -- /md/lint -i _releasenotes .
 
 .PHONY: lint-shell
-lint-shell: ## Lint the project's shell scripts
-	docker run --rm -t -v "$$(pwd)":/build:ro gcr.io/cluster-api-provider-vsphere/extra/shellcheck
+lint-shell: verify-shellcheck ## Lint the project's shell scripts
```
Comment (sbueringer) [inline Makefile:331]: Should we get rid of lint-shell?

Comment (chrischdi) [reply Makefile:331]: I'd be okay with that 👍

Comment (sbueringer) [reply Makefile:331]: Let's do it

Comment (chrischdi) [reply Makefile]: Also verified we do not use this or the shell script in CI.

**Comment by sbueringer** [comment]
There's not much to hold at this point :)

**Pull request merged**